admin-dev/themes/new-theme/js/components/modal/progress-modal.ts

```diff
@@ -0,0 +1,421 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+/* eslint max-classes-per-file: ["error", 2] */
+
+import {
+  ModalContainerType, ModalContainer, ModalType, ModalParams, Modal, InputModalParams,
+} from '@components/modal/modal';
+
+export interface ProgressModalParentContainerType extends ModalContainerType {
+}
+export interface ProgressModalType extends ModalType {
+  modal: ProgressModalParentContainerType;
+}
+export type ProgressModalParams = ModalParams & {
+  modalTitle: string;
+  total: number;
+  confirmCallback: (event: Event) => void,
+  customButtons: Array<HTMLButtonElement | HTMLAnchorElement>;
+}
+export type InputProgressModalParams = Partial<ProgressModalParams>;
```

Comment:
```suggestion
export type InputProgressModalParams = Partial<ProgressModalParams> & {
    total: number;
};
```

Reply:
This way total is a mandatory parameter even for input, no default value will be provided